### PR TITLE
Fix module EctoValidator

### PR DESCRIPTION
**Pull request by eduardoscottini**

### CODE REVIEW

File: lib/ecto_validator.ex [hunk 1]
```diff
@@ -22,62 +22,33 @@ defmodule Fields.EctoValidator do
...
-    |> Validate.postcode()
+    |> valid?(validator)
     |> case do
       true -> changeset
       false -> add_error(changeset, field, message)
     end
   end
+
+  defp valid?(nil, _validator), do: true
+  defp valid?(value, validator), do: apply(validator, [value])
```
Comment (gabrielpra1) [inline lib/ecto_validator.ex]:
```suggestion
  defp valid?(value, validator), do: validator.(value)
```

File: config/test.exs [hunk 2]
```diff
@@ -5,3 +5,18 @@ config :fields, Fields,
 
 config :fields, Fields.AES,
   key: "i6uIb0FMGd8fouIDd/4qkf6LhI6PG/ZKpHBEN2ed5Mk=" |> :base64.decode
+
+# Repos known to Ecto:
+config :fields, ecto_repos: [Fields.Repo]
+
+# Test Repo settings
+config :fields, Fields.Repo,
+database: "fields_test",
```
Comment (gabrielpra1) [inline config/test.exs]: add indentation to these configs

File: priv/repo/migrations/20191126115123_create_tables.exs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+
+defmodule Crudry.Repo.Migrations.CreateTables do
```
Comment (gabrielpra1) [inline priv/repo/migrations/20191126115123_create_tables.exs]: Do we need postgres and migrations to test the changeset? I think an embedded schema could be used instead

**Review by gabrielpra1 — CHANGES_REQUESTED** [review]

File: mix.lock [hunk 4]
```diff
@@ -1,8 +1,13 @@
...
+  "ecto_sql": {:hex, :ecto_sql, "3.2.2", "d10845bc147b9f61ef485cbf0973c0a337237199bd9bd30dd9542db00aadc26b", [:mix], [{:db_connection, "~> 2.1", [hex: :db_connection, repo: "hexpm", optional: false]}, {:ecto, "~> 3.2.0", [hex: :ecto, repo: "hexpm", optional: false]}, {:myxql, "~> 0.2.0 or ~> 0.3.0", [hex: :myxql, repo: "hexpm", optional: true]}, {:postgrex, "~> 0.15.0", [hex: :postgrex, repo: "hexpm", optional: true]}, {:telemetry, "~> 0.4.0", [hex: :telemetry, repo: "hexpm", optional: false]}], "hexpm"},
   "elixir_make": {:hex, :elixir_make, "0.6.0", "38349f3e29aff4864352084fc736fa7fa0f2995a819a737554f7ebd28b85aaab", [:mix], [], "hexpm"},
+  "postgrex": {:hex, :postgrex, "0.15.1", "23ce3417de70f4c0e9e7419ad85bdabcc6860a6925fe2c6f3b1b5b1e8e47bf2f", [:mix], [{:connection, "~> 1.0", [hex: :connection, repo: "hexpm", optional: false]}, {:db_connection, "~> 2.1", [hex: :db_connection, repo: "hexpm", optional: false]}, {:decimal, "~> 1.5", [hex: :decimal, repo: "hexpm", optional: false]}, {:jason, "~> 1.0", [hex: :jason, repo: "hexpm", optional: true]}], "hexpm"},
```
Comment (gabrielpra1) [inline mix.lock]: the dependencies are still here, run `mix deps.unlock --unused` to remove them

**Review by gabrielpra1 — CHANGES_REQUESTED** [review]

File: mix.lock (hunk 4 above)
Comment (eduardoscottini) [reply mix.lock]: Done

**Pull request merged**